### DISCUSSION TITLE
Keycloak: Does/Can LDAPStorageMapper have a onDeleteUserFromLDAP override method?

### DISCUSSION THREAD

**Discussion by vineethcsreepad** [keycloak/keycloak#17507] [Q&A]
UseCase: We have implemented custom keycloak LDAP Storage Mapper for LDAP/User Federation Sync. Followed the same aritcle as pasted here : https://medium.com/@ivancheahkf/keycloak-event-listener-spi-for-ldap-user-federation-sync-62fa17c573bc

This works absolutely fine for a new user added/updated to AD and a periodic sync will trriger the override method "onImportUserFromLDAP" in LDAPStorageMapper. We have implemented the same in our POC and have tested it as well.

Curious to know if there is anything similar if/when a user is deleted from AD and syncd to keycloak, any override method exists user LDAPStorageMapper.

Currently we see this message "Sync changed users finished: 0 imported users, 0 updated users" after a sync from AD keycloak is initiated. User is also deleted from keycloak, never we which override method is called onDelete.

**Comment by mrpotato3** [comment]
+1
